### ISSUE TITLE
<S-Insert> clobbers paste

### ISSUE THREAD

**Issue by Tanath** [tridactyl/tridactyl#2414]
Tridactyl version: 1.19.0pre3821-f4fa5b8
Firefox version: Mozilla Firefox 76.0.1
Operating system: linux

*   Brief description of the problem:  
`<S-Insert>` is a standard/universal shortcut to paste (paired with `<C-Insert>` to copy). Tridactyl clobbers it by default, binding it to ignore mode.

**Comment by glacambre** [comment]
I agree that this binding is bad. I think it was introduced because OSX 
users that have a touchbar can't press <S-Esc> or something. It's 
probably too late to remove it (although perhaps we could only create it 
on OSX, with a config updater?).

I use `bind --mode=insert <C-V> composite getclip selection | 
text.insert_text` instead and it's pretty nice, but it does have the 
disadvantage of only working in tridactyl.

**Comment by bovine3dom** [comment]
> I think it was introduced because OSX users that have a touchbar can't press <S-Esc> or something.

No, that's ctrl-alt-backtick. Macs don't have sensible insert keys either.

This is my fault. I thought the bind in Vimperator was shift-insert but it is actually just insert. It has been in Tridactyl for so long that we can't really remove it. Maybe something for V2: #434. 

You can, of course, unbind it with `unbind <S-Insert>` and `unbind ---mode=ignore <S-Insert>`.

**Comment by Tanath** [comment]
> You can, of course, unbind it with `unbind <S-Insert>` and `unbind ---mode=ignore <S-Insert>`.

`unbind <S-Insert>` is what I'm trying to do - prevented by #2415 - do I also need `unbind --mode=ignore <S-Insert>`? If so, why both?

**Comment by glacambre** [comment]
On 2020-05-24 16:34, Mike Smith wrote:
> do I also need |unbind --mode=ignore <S-Insert>|
I believe so, yes. This is because in normal mode <S-Insert> is bound to 
`:mode ignore` and in ignore more it's bound to `:mode normal`, to allow 
you to toggle ignore mode with a single key.

**Comment by bovine3dom** [comment]
I don't understand how #2415 would prevent you from unbinding anything - Tridactyl has persistent storage that should "just work" unrelated to the RC file.

The first unbind has an implicit `--mode=normal`. Shift insert "toggles" ignore mode simply by being bound to enter ignore mode in normal mode, and being bound to enter normal mode in ignore mode. So if you want to paste in ignore mode you'll need it.

**Comment by Tanath** [comment]
OK. Yes, I'm able to unbind, but I thought it was temporary since I can't save config. I hadn't actually tested that yet. I've been avoiding further changes until `:mkt` works.

I would recommend you stop recommending `<S-Insert>` on the new tab page and in help. I see there's alternative shortcuts to suggest.

**Comment by bovine3dom** [comment]
Ah, I think I have a better idea:

- `<Insert>: "mode ignore"` etc and update help pages
- `<S-Insert>: "fillcmdline <S-Insert> to toggle ignore mode is deprecated to allow it to be used for paste. Use <Insert> or :bind <S-Insert> --mode [mode] mode ignore"`
- add a warning to the new tab page
- release 1.20.0
- in 1.21.0 remove the bind and the deprecation warning.

> I thought it was temporary since I can't save config.

I hear this quite a bit from people. How/where could we explain this better?

**Comment by Tanath** [comment]
> I hear this quite a bit from people. How/where could we explain this better?

It's the default assumption, I think. I would explain how configs work in the `mktridactylrc` section of `:help`.

**Comment by bovine3dom** [comment]
Done - https://github.com/tridactyl/tridactyl/commit/ced64e8fb689a0a4efb5cdccfa0ff642965e602b - thanks :)
